paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt

```diff
@@ -91,6 +93,18 @@ internal class PaymentOptionsViewModel @Inject constructor(
             PaymentOptionResult.Canceled(mostRecentError = _fatal.value)
     }
 
+    override fun onFinish() {
+        onUserSelection()
+    }
+
+    override fun onError(@StringRes error: Int?) {
+        error?.let {
+            onFatal(
```

Comment:
> A couple of test cases to check:
> 
> 1. Verify the save for future use checkbox actually impacts if the card is actually saved on a returning user?
> 2. Verify the checkbox is not shown if it is a setup intent or if P&S (aka payment intent with setup for future usage).
> 3. Verify the following sequence:
>    a. open with a returning user
>    b. Add a new LPM, and select us bank account
>    c. Fill in all the fields
>    d. attach it.
>    d. Return to the returning user selected fields
>    e. Open the us bank account again.
>    I think there are many sane behaviors here including that the US bank account is blank when you return, and you can choose a new card to re-attach.  (Make sure the save for future usage checkbox is restored to a correct state as well).
> 4. Just make sure with don't keep activities turned on the activity is restored correctly before attach, after attach.
> 5. Check the screen and values when accessibility is turned on.  Our Onboarding guide links to some good accessibility test strategies.


✅ Verify the save for future use checkbox actually impacts if the card is actually saved on a returning user?
✅ Verify the checkbox is not shown if it is a setup intent or if P&S (aka payment intent with setup for future usage).
✅ Verify the following sequence
- The PM will be attached after pressing "Pay", so the customer won't be able to view the form after attaching.

❌ Just make sure with don't keep activities turned on the activity is restored correctly before attach, after attach.
- This will be a follow up PR

✅ Check the screen and values when accessibility is turned on.  Our Onboarding guide links to some good accessibility test strategies.